relnotes/reverse.feature.md

```diff
@@ -0,0 +1,5 @@
+* `ocean.core.array.Iteration`
+
+  New module containing single function for now - `reverse`, which allows to
+  iterate given slice in reverse order without any allocations (similar to
+  built-in `foreach_reverse`).
```

Comment:
Because library version is better? For example, it only works on arrays and does not silently accept `opApply` struct pretending to reverse it when it doesn't. I'd like to introduce a static lint check one day that prohibits usage of `foreach_reverse` completely tbh.